### ISSUE TITLE
Finding mumps DLLs on MS Windows

### ISSUE THREAD

**Issue by zitkat** [sfepy/sfepy#608]
I have tried using mumps as linear solver on MS Windows. The easiest way of obtainin it seemed to me to install it from conda: https://anaconda.org/conda-forge/mumps. Sfepy searches for the DLLs with name `[libname].dll` where "liname" is name of the solver component, however conda installs them as `lib[libname].dll`. This is the same naming convention as on linux systems. In linked PR I propose fallback to `lib` prefix for Windows when the library is not found at first.

**Comment by zitkat** [comment]
https://github.com/sfepy/sfepy/pull/609

**Comment by rc** [comment]
FYI: I am having troubles as well - I installed it using conda indirectly by installing/updating `petsc4py`, then `mumps` segfaults:
```
$ python3 ./simple.py examples/linear_elasticity/linear_elastic.py
...
sfepy: nls: iter: 0, residual: 4.778054e-03 (rel: 1.000000e+00)
*** Error in `python3': corrupted size vs. prev_size: 0x0000564aa9da72b0 ***
```
When I install it via the `mumps`package, there is no segfault, but the solver (mumps-5.1.2) does not work:
```
sfepy: nls: iter: 0, residual: 4.778054e-03 (rel: 1.000000e+00)
sfepy:   residual:    0.02 [s]
sfepy:     matrix:    0.02 [s]
sfepy:      solve:    0.02 [s]
sfepy: warning: linear system solution precision is lower
sfepy: then the value set in solver options! (err = 3.925994e-03 < 1.000000e-10)
sfepy: nls: iter: 1, residual: 3.925994e-03 (rel: 8.216723e-01)
```
The "system" installed mumps-4.10.0 works fine.

Pinging @vlukes, there might be some API changes in the mumps library that are not yet reflected in our wrappers.

**Comment by vlukes** [comment]
:+1:

**Comment by vlukes** [comment]
I have mumps-5.1.2 installed via `apt` on my ubuntu-based system and  the solver works well:

`./simple.py examples/linear_elasticity/linear_elastic.py`

gives
```
sfepy: ...done in 0.00 s
sfepy: nls: iter: 0, residual: 4.778054e-03 (rel: 1.000000e+00)
sfepy:   residual:    0.01 [s]
sfepy:     matrix:    0.01 [s]
sfepy:      solve:    0.04 [s]
sfepy: nls: iter: 1, residual: 6.302540e-18 (rel: 1.319060e-15)
sfepy: solved in 1 steps in 0.07 seconds
```

**Comment by rc** [comment]
Thanks for checking it. It is wierd, could you try conda as well? I wonder what is the problem (ctypes version, ...?).
